programs/cleos/program_location.hpp

```diff
@@ -0,0 +1,65 @@
+#pragma once
+#include <boost/predef/os.h>
+#include <filesystem>
+
+// adapted from boost source boost/dll/detail/posix/program_location_impl.hpp
```

Comment:
Can we continue to use boost DLL with `BOOST_DLL_USE_STD_FS`, so we don't need this local reimplementation?